pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py

```diff
@@ -122,11 +122,10 @@ def _VSCODE_getRowCount(var):
 
 # Function to retrieve a set of rows for a data frame
 def _VSCODE_getDataFrameRows(df, start, end):
-    df = _VSCODE_convertToDataFrame(df)
+    df = _VSCODE_convertToDataFrame(df[start:end])
```

Comment:
This makes viewing N-D data 30% faster (previously we were flattening the whole variable and then throwing everything except the requested rows away)